inc/zoo/swar/SWAR.h

```diff
@@ -472,13 +481,53 @@ booleans(SWAR<NB, T> arg) noexcept {
 template<int NBits, typename T>
 constexpr auto
 differents(SWAR<NBits, T> a1, SWAR<NBits, T> a2) {
-    return booleans(a1 ^ a2);
+    return ~equals(a1, a2);
 }
 
+/**
+ * @return BooleanSWAR that contains a true value in each lane where the two
+ * inputs SWARs match.
+ */ 
+template<int NBits, typename T>
+constexpr BooleanSWAR<NBits, T>
```

Comment:
I prefer `auto` here, especially since you explicitly build a `BooleanSWAR` at the return site.
I have no problem to leave here the explicit return type, but then return `{ expression }`.

Reply:
Is it not forced due to the usage of 'friend'?
I also prefer auto here.

Reply:
Ok, this combined with the other type thing is a compile error side quest I got distracted from eventually fixing, thanks for noticing.
Seems like forcing a BooleanSWAR return isn't best with all auto (since that forces a mask that might be unnecessary)

Reply:
What unnecessary mask?
Do you mean the masking to convert a `SWAR` to a `BooleanSWAR`?
I do not think about it because in the context of a return value the optimizers will readily identify whether the masking in the conversion is redundant.